File: LLM-Translations/GPT-4/codenet/C/Java/s582614433.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.Scanner;

class Main {
    static class Edge {
        long p;
        int i;
        String s;

        Edge(String s, long p, int i) {
            this.s = s;
            this.p = p;
            this.i = i;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Edge[] data = new Edge[n];

        for (int i = 0; i < n; i++) {
            String s = sc.next();
            long p = sc.nextLong();
            data[i] = new Edge(s, p, i);
        }

        Arrays.sort(data, new Comparator<Edge>() {
            @Override
            public int compare(Edge e1, Edge e2) {
                int sCompare = e1.s.compareTo(e2.s);
                if (sCompare != 0) return sCompare;
                if (e1.p < e2.p) return 1;
                return -1;
            }
        });

        for (int i = 0; i < n; i++) {
            System.out.println(data[i].i + 1);
        }
    }
}
// 

File: LLM-Translations/GPT-4/codenet/C/Python/s582614433.py

from functools import cmp_to_key

class Edge:
    def __init__(self, p, i, s):
        self.p = p
        self.i = i
        self.s = s

def esort(e1, e2):
    if e1.s != e2.s:
        return (e1.s > e2.s) - (e1.s < e2.s)
    if e1.p < e2.p:
        return 1
    return -1

n = int(input())
data = []

for i in range(n):
    s, p = input().split()
    p = int(p)
    data.append(Edge(p, i, s))

data.sort(key=cmp_to_key(esort))

for i in range(n):
    print(data[i].i + 1)

#
